pkg/policy/rule.go

```diff
@@ -54,60 +51,18 @@ func mergeL4IngressPort(ctx *SearchContext, endpoints []api.EndpointSelector, r
 	// for merging with the filter which is already in the policy map.
 	filterToMerge := CreateL4IngressFilter(endpoints, r, p, proto, ruleLabels)
 
-	// Handle cases where filter we are merging new rule with, or new rule itself
-	// allows all traffic on L3.
+	// Handle cases where filter we are merging new rule with, new rule itself
+	// allows all traffic on L3, or both rules allow all traffic on L3.
 	//
-	// Existing filter selects all endpoints, so don't add new endpoints.
-	if existingFilter.AllowsAllAtL3() && len(filterToMerge.Endpoints) > 0 {
-		log.WithFields(logrus.Fields{
-			logfields.EndpointSelector: filterToMerge.Endpoints,
-			"policy":                   existingFilter,
-		}).Debug("skipping L4 filter as the endpoints are already covered")
-
-		// Existing L4Filter already selects all endpoints; if there are no L7 rules
-		// to add from the new rule, then we can just exit because the rule is as
-		// permissive as possible at L3, the L4 information is already contained
-		// within the filter, and there is no L7 metadata to add to the filter.
-		if r.NumRules() == 0 {
-			existingFilter.DerivedFromRules = append(existingFilter.DerivedFromRules, ruleLabels)
-			resMap[key] = existingFilter
-			return 1, nil
-		}
+	// Case 1: Both filters select all endpoints
```

Comment:
Fix the comment. This doesn't match the test below.

Same in `mergeL4EgressPort`.